scripts/mech_events.py

```diff
@@ -0,0 +1,266 @@
+        self.block_number = block_number
+        self.ipfs_link = ""
+        self.ipfs_contents = {}
+        self._populate_ipfs_contents(data)
+
+    def _populate_ipfs_contents(self, data: str) -> None:
+        url = f"{IPFS_ADDRESS}{CID_PREFIX}{data}"
+        for _url in [f"{url}/metadata.json", url]:
+            try:
+                response = requests.get(_url)
+                response.raise_for_status()
+                self.ipfs_contents = response.json()
+                self.ipfs_link = _url
+            except Exception:  # pylint: disable=broad-except
+                continue
+
+
+@dataclass
+class MechRequest(MechBaseEvent):
+    """A mech's on-chain response representation."""
+
+    request_id: str
+    fee: int
+    event_name: str = "Request"
+
+    def __init__(self, event: AttributeDict):
+        """Initializes the MechRequest"""
+
+        if self.event_name != event["event"]:
+            raise ValueError("Invalid event to initialize MechRequest")
+
+        args = event["args"]
+
+        super().__init__(
+            event_id=args["requestId"],
+            data=args["data"].hex(),
+            sender=args["sender"],
+            transaction_hash=event["transactionHash"].hex(),
+            block_number=event["blockNumber"],
+        )
+
+        self.request_id = self.event_id
+        self.fee = 10000000000000000
+
+
+@dataclass
+class MechDeliver(MechBaseEvent):
+    """A mech's on-chain response representation."""
+
+    event_name: str = "Deliver"
+
+
+def _read_mech_events_data_from_file() -> Dict[str, Any]:
+    """Read Mech events data from the JSON file."""
+    try:
+        with open(MECH_EVENTS_JSON_PATH, "r", encoding="utf-8") as file:
+            mech_events_data = json.load(file)
+    except FileNotFoundError:
+        mech_events_data = {}
+    return mech_events_data
+
+
+# pylint: disable=too-many-locals
+def _update_mech_events_db(
+    rpc: str,
+    mech_contract_address: str,
+    event_name: str,
+    earliest_block: int,
+    sender: str,
+) -> None:
+    """Get the mech Request events."""
+
+    print(
+        f"Updating the local Mech events database. This may take a while.\n"
+        f"           Event: {event_name}\n"
+        f"   Mech contract: {mech_contract_address}\n"
+        f"  Sender address: {sender}"
+    )
+
+    # Read the current Mech events database
+    mech_events_data = _read_mech_events_data_from_file()
+
+    # Search for Mech events in the blockchain
+    try:
+        w3 = Web3(HTTPProvider(rpc))
+        with open(AGENT_MECH_JSON_PATH, "r", encoding="utf-8") as file:
+            contract_data = json.load(file)
+
+        abi = contract_data.get("abi", [])
+        contract_instance = w3.eth.contract(address=mech_contract_address, abi=abi)
+
+        last_processed_block = (
+            mech_events_data.get(sender, {})
+            .get(mech_contract_address, {})
+            .get(event_name, {})
+            .get("last_processed_block", 0)
```

Comment:
Maybe it makes sense to use the block from which the contracts were deployed.

Reply:
This must be changed in lines 57 and 62.

Reply:
Fixed